addon/mixins/data-adapter-mixin.js

```diff
@@ -30,10 +30,14 @@ export default Ember.Mixin.create({
   session: service('session'),
 
   ajaxOptions() {
+    const authorizer = this.get('authorizer');
     let hash = this._super(...arguments);
     let { beforeSend } = hash;
+
+    Ember.assert("You're using the DataAdapterMixin without specifying an authorizer. Please add `authorizer: 'authorizer:application'` to your adapter.", Ember.isPresent(authorizer));
+
```

Comment:
What are your thoughts on defaulting `authorizer` to `authorizer:application`,  then checking with the container for whether it exists? If it doesn't, then assert that this needs customising?